CMakeLists.txt

```diff
@@ -26,58 +26,99 @@
+    OpenVDB Grid ABI, where N is 3, 4, 5 etc. (some newer features will be disabled)")
 
-IF (OPENVDB_BUILD_HOUDINI_SOPS)
+IF ( OPENVDB_BUILD_HOUDINI_SOPS )
+  # @todo why?
   SET ( OPENVDB_BUILD_PYTHON_MODULE OFF )
 ENDIF ()
 
+# Some platform checks
+
 IF ( APPLE )
   IF ( OPENVDB_BUILD_HOUDINI_SOPS AND OPENVDB_BUILD_MAYA_PLUGIN )
-	MESSAGE ( FATAL_ERROR "On OSX, Houdini linked with libc++ whereas Maya is linked with libstdc++, the different runtimes are not mutually compatible, build them Houdini SOPs and Maya plugins separately with the requisite DCC's compiler" )
+  MESSAGE ( FATAL_ERROR "On OSX, Houdini linked with libc++ whereas Maya is linked with libstdc++, "
+        "the different runtimes are not mutually compatible, build them Houdini SOPs and Maya plugins "
+        "separately with the requisite DCC's compiler" )
   ENDIF ()
 ENDIF ()
 
-LIST(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
-
-SET ( OPENVDB_TOP_LEVEL_DIR ${PROJECT_SOURCE_DIR} CACHE PATH "OpenVDB Top Level Directory to share among sub projects" )
+# Determine OpenVDB version
+
+SET ( OPENVDB_VERSION_FILE ${CMAKE_CURRENT_SOURCE_DIR}/openvdb/version.h )
+FILE ( STRINGS "${OPENVDB_VERSION_FILE}" openvdb_major_version_str REGEX "^#define[\t ]+OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER[\t ]+.*")
+FILE ( STRINGS "${OPENVDB_VERSION_FILE}" openvdb_minor_version_str REGEX "^#define[\t ]+OPENVDB_LIBRARY_MINOR_VERSION_NUMBER[\t ]+.*")
+FILE ( STRINGS "${OPENVDB_VERSION_FILE}" openvdb_patch_version_str REGEX "^#define[\t ]+OPENVDB_LIBRARY_PATCH_VERSION_NUMBER[\t ]+.*")
+
+STRING (REGEX REPLACE "^.*OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER[\t ]+([0-9]*).*$" "\\1"
+  _openvdb_major_version_number "${openvdb_major_version_str}")
+STRING (REGEX REPLACE "^.*OPENVDB_LIBRARY_MINOR_VERSION_NUMBER[\t ]+([0-9]*).*$" "\\1"
+  _openvdb_minor_version_number "${openvdb_minor_version_str}")
+STRING (REGEX REPLACE "^.*OPENVDB_LIBRARY_PATCH_VERSION_NUMBER[\t ]+([0-9]*).*$" "\\1"
+  _openvdb_patch_version_number "${openvdb_patch_version_str}")
+
+SET ( OPENVDB_MAJOR_VERSION_NUMBER ${_openvdb_major_version_number} CACHE STRING "OpenVDB Major Version Number" )
+SET ( OPENVDB_MINOR_VERSION_NUMBER ${_openvdb_minor_version_number} CACHE STRING "OpenVDB Minor Version Number" )
+SET ( OPENVDB_PATCH_VERSION_NUMBER ${_openvdb_patch_version_number} CACHE STRING "OpenVDB Build Version Number" )
+
+# Determine the OpenVDB ABI Version. If OPENVDB_ABI_VERSION_NUMBER is defined, use that value,
+# otherwise check the legacy OPENVDB_ENABLE_3_ABI_COMPATIBLE option, falling back to the major
+# library version number
+MESSAGE ( STATUS "Using OpenVDB ABI Version ${OPENVDB_ABI_VERSION_NUMBER}" )
+
+IF ( OPENVDB_ABI_VERSION_NUMBER STREQUAL "" )
+  IF ( OPENVDB_ENABLE_3_ABI_COMPATIBLE )
+    SET ( OPENVDB_ABI_VERSION_NUMBER 3 )
+    ADD_DEFINITIONS ( -DOPENVDB_3_ABI_COMPATIBLE )
+    MESSAGE ( DEPRECATION "OPENVDB_ENABLE_3_ABI_COMPATIBLE is deprecated. Instead use
```

Comment:
Although `OPENVDB_ENABLE_3_ABI_COMPATIBLE` is deprecated, until it is removed it would probably be best to add `-DOPENVDB_3_ABI_COMPATIBLE` if `OPENVDB_ABI_VERSION_NUMBER` is 3.

Reply:
Done